ui/pages/confirmations/components/confirm/header/header.tsx

```diff
@@ -26,24 +26,20 @@ import { DAppInitiatedHeader } from './dapp-initiated-header';
 import HeaderInfo from './header-info';
 import { WalletInitiatedHeader } from './wallet-initiated-header';
 
+const CONFIRMATIONS_WITH_NEW_HEADER = [
+  TransactionType.tokenMethodTransfer,
+  TransactionType.tokenMethodTransferFrom,
+  TransactionType.tokenMethodSafeTransferFrom,
+];
+
 const Header = () => {
   const { networkImageUrl, networkDisplayName } = useConfirmationNetworkInfo();
   const { senderAddress: fromAddress, senderName: fromName } =
     useConfirmationRecipientInfo();
 
   const { currentConfirmation } = useConfirmContext<Confirmation>();
 
-  if (currentConfirmation?.type === TransactionType.tokenMethodTransfer) {
-    const isWalletInitiated =
-      (currentConfirmation as TransactionMeta).origin === 'metamask';
-
-    if (isWalletInitiated) {
-      return <WalletInitiatedHeader />;
-    }
-    return <DAppInitiatedHeader />;
-  }
-
-  return (
+  const DefaultHeader = (
```

Comment:
Minor, we've defined this as a nested component within the `Header` component but is it slightly more readable to define it  at the top level and either re-use the same hooks or pass in the properties it needs?